src/csharp/Microsoft.Spark/Sql/Column.cs

```diff
@@ -592,6 +592,16 @@ public Column As(IEnumerable<string> alias)
             return Apply("as", alias);
         }
 
+        /// <summary>
+        /// Returns a column with values modified as per input object.
+        /// </summary>
+        /// <param name="colObject">Column object to apply</param>
+        /// <returns>Column object</returns>
+        public Column Apply(object colObject)
```

Comment:
Is this only going to be exposed for Column type or anything like how scala has it defined

```scala
  /**
   * Extracts a value or values from a complex type.
   * The following types of extraction are supported:
   * <ul>
   * <li>Given an Array, an integer ordinal can be used to retrieve a single value.</li>
   * <li>Given a Map, a key of the correct type can be used to retrieve an individual value.</li>
   * <li>Given a Struct, a string fieldName can be used to extract that field.</li>
   * <li>Given an Array of Structs, a string fieldName can be used to extract filed
   *    of every struct in that array, and return an Array of fields.</li>
   * </ul>
   * @group expr_ops
   * @since 1.4.0
   */
  def apply(extraction: Any): Column = withExpr {
    UnresolvedExtractValue(expr, lit(extraction).expr)
  }
```

Reply:
Any which is why it is an object, I will change the parameter name to avoid coincidence

Reply:
The summary description should be updated to be somewhat similar to scala's.